lib/api/api.go

```diff
@@ -386,13 +382,14 @@ func (s *service) Serve(ctx context.Context) error {
 	handler = withDetailsMiddleware(s.id, handler)
 
 	// Wrap everything in auth, if user/password is set or WebAuthn is enabled.
-	if s.IsAuthEnabled() {
-		authMW := newBasicAuthAndSessionMiddleware(&s.tokenCookieManager, guiCfg, s.cfg.LDAP(), handler, s.evLogger)
+	if s.isAuthEnabled(&webauthnService, guiCfg) {
```

Comment:
Lets just return a pointer to the service from the constructor in the first place. Seems a bit inconsistent to care about copying in some places but not others.